File: Baekjoon/Class4/07.15666 N과 M (12).js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

const [_, m] = input[0].split(" ").map(Number);
const track = [];
const nums = Array.from(
    new Set(input[1].split(" ").map(Number)) // 중복 제거
).sort((a, b) => a - b); // 정렬
const out = [];

const backTrack = (start) => {
    if(track.length === m) { // 출력
        out.push(track.join(" "));
        return;
    }
    for(let i = start; i < nums.length; i++) {
        track.push(nums[i]); // 추가
        backTrack(i);
        track.pop(); // 삭제
    }
}

backTrack(0);
console.log(out.join("\n"));

File: Baekjoon/Class4/07.15666 N과 M (12).py
import sys
input = sys.stdin.readline
write = sys.stdout.write

_, m = map(int, input().split())
track = []
nums = list(map(str, sorted(set(map(int, input().split()))))) #중복 제거, 정렬

def back_track(start):
    if len(track) == m: #출력
        write(f"{" ".join(track)}\n")
        return
    for i in range(start, len(nums)):
            track.append(nums[i]) #추가
            back_track(i)
            track.pop() #삭제

back_track(0)